#Question: What percentage of male first-class passengers survived?

We are conducting a retrospective observational census. We are conducting the study through a census because it allows us to have access to all the data, leading to better results. The data I am using is not complete, which may lead to biases. Furthermore, I don't know how the data was collected, so there may have been bias in the data collection as well.

In [6]:
import pandas

In [7]:
df = pandas.read_csv('titanic.csv')

In [8]:
survived = len(df[(df["Pclass"] == 1) & (df["Sex"] == "male") & (df["Survived"] == 1)])
print(survived)

45


In [9]:
died = len(df[(df["Pclass"] == 1) & (df["Sex"] == "male") & (df["Survived"] == 0)])
print(died)

77


In [10]:
str(survived / died) + "%"

'0.5844155844155844%'

45 out of 77 male first-class passengers survived the sinking of the Titanic, giving a survival rate of 58.442%. Some possible sources of bias may be the unknown origins of the data and the incomplete data set, possibly leading to skewed data